haas/client/node.py

```diff
@@ -0,0 +1,121 @@
+                    "No such node exist. Nothing to delete."
+                    )
+
+    def power_cycle(self, node_name):
+        """ Power cycles the <node> """
+        self.node_name = node_name
+        url = self.object_url('node', node_name, 'power_cycle')
+        q = self.s.post(url)
+        if q.ok:
+            return
+        elif q.status_code == 409:
+            raise errors.BlockedError(
+                    "Operation blocked by other pending operations"
+                    )
+        elif q.status_code == 500:
+            raise errors.NotFoundError(
+                    "Operation Failed. Contact your system administrator"
+                    )
+
+    def power_off(self, node_name):
+        """ Power offs the <node> """
+        self.node_name = node_name
+        url = self.object_url('node', node_name, 'power_off')
+        q = self.s.post(url)
+        if q.ok:
+            return
+        elif q.status_code == 404:
+            raise errors.NotFoundError("Node not found.")
+        elif q.status_code == 409:
+            raise errors.BlockedError(
+                    "Operation blocked by other pending operations"
+                    )
+        elif q.status_code == 500:
+            raise errors.NotFoundError(
+                    "Operation Failed. Contact your system administrator"
+                    )
+
+    def add_nic(self, node_name, nic_name, macaddr):
+        """ adds a <nic> from <node>"""
+        self.node_name = node_name
+        self.nic_name = nic_name
+        self.macaddr = macaddr
+        url = self.object_url('node', node_name, 'nic', nic_name)
+        payload = json.dumps({'macaddr': macaddr})
+        q = self.s.put(url, data=payload)
+        if q.ok:
+            return
+        elif q.status_code == 404:
+            raise errors.NotFoundError(
+                    "Nic cannot be added. Node does not exist."
+                    )
+        elif q.status_code == 409:
+            raise errors.DuplicateError(
+                    "Nic already exists."
+                    )
+
+    def remove_nic(self, node_name, nic_name):
+        """ remove a <nic> from <node>"""
+        self.node_name = node_name
+        self.nic_name = nic_name
+        url = self.object_url('node', node_name, 'nic', nic_name)
+        q = self.s.delete(url)
+        if q.ok:
+            return
+        elif q.status_code == 404:
+            raise errors.NotFoundError(
+                    "Nic not found. Nothing to delete."
+                    )
+        elif q.status_code == 409:
+            raise errors.BlockedError(
+                    "Cannot delete nic, diconnect it from network first"
+                    )
+
+    def connect_network(self, node_name, nic, network, channel):
+        """ Connect <node> to <network> on given <nic> and <channel>"""
+        pass
+
+    def disconnect_network(self, node_name, nic, network):
+        """ Disconnect <node> from <network> on the given <nic>. """
+        pass
```

Comment:
for this PR, I do not wish to add more calls. Each calls means changing equivalent code in CLI and writing tests. In future I will make smaller PR requests for these and other missing calls.